### PR TITLE
feat(source_version): add SOURCE_VERSION env var

### PR DESCRIPTION
**Pull request by aboyett**
heroku exposes the git commit hash of the revision being built in the
SOURCE_VERSION environment variable. this commit implements this in the
deis builder for git pushes.

see https://devcenter.heroku.com/changelog-items/630 for more info on
heroku's implementation

### CODE REVIEW

**Comment by deis-admin** [comment]
Thanks for the contribution! Please ensure your commits follow our [style guide](http://docs.deis.io/en/latest/contributing/standards/#commit-style-guide). This code will be tested once a Deis maintainer reviews it.

**Comment by jeroenvisser101** [comment]
@aboyett looks good, maybe add some unit-tests?

**Comment by aboyett** [comment]
@jeroenvisser101 I augmented the existed tests to handle this new argument. Do you have any suggestions on what other tests you'd like to see added as part of this PR?

**Comment by jeroenvisser101** [comment]
@aboyett ah, right, I think it's fine indeed. In #412 I added a method, but you're reusing `gitSha.short()` which has tests already. I think it's fine as is.

File: pkg/gitreceive/build.go [hunk 1]
```diff
@@ -167,6 +167,7 @@ func build(
 			conf.PodNamespace,
 			appConf.Values,
 			slugBuilderInfo.TarKey(),
+			gitSha.Short(),
```
Comment (kmala) [inline pkg/gitreceive/build.go]: this is not required as it is for docker file builds.

Comment (jeroenvisser101) [inline pkg/gitreceive/build.go]: The env is added to the dockerBuilderPod as well, it might be useful for someone?

**Comment by Joshua-Anderson** [comment]
Jenkins, add to whitelist.

File: pkg/gitreceive/build.go (hunk 1 above)
Comment (aboyett) [inline pkg/gitreceive/build.go]: You're right that it isn't required for docker builds, but I thought that providing parity is useful as it means the `SOURCE_VERSION` variable exists in the build env whenever code is built as a result of a `git push deis`. I will certainly be using this extra functionality but I am happy to break it out into a separate commit or separate PR for further discussion, let me know.

**Comment by Joshua-Anderson** [comment]
@aboyett Some other changes have been made to this bit of code, do you mind rebasing?

**Comment by aboyett** [comment]
Rebase complete, waiting on the e2e test to finish

**Comment by felixbuenemann** [comment]
If I understand this PR correctly, the SOURCE_VERSION environment variable is currently only available during build. Would it be possible to pass it on to the apps environment as well?

It would be great to pass as additional context in exception monitoring. I'm currently passing the WORKFLOW_RELEASE environment var as context, but that requires another roundtrip to look up the commit sha from `deis releases` to be able to see which version of the code this was.

**Comment by bacongobbler** [comment]
@felixbuenemann mind filing a new issue so we can track that effort? :)

**Comment by aboyett** [comment]
My pull request attempted to replicate the behavior of Heroku which only makes the `SOURCE_VERSION` variable available during the compile phase[1]. I am curious to hear more about this `HEROKU_SLUG_COMMIT` env var as it doesn't appear to exist in cedar-14 based python buildpack generated slugs. If you create a new ticket, please reference this ticket so it is easy to find :smile:

[1] https://devcenter.heroku.com/articles/buildpack-api#bin-compile-summary

**Comment by felixbuenemann** [comment]
@aboyett It must be enabled on heroku, see [Heroku Labs: Dyno Metadata](https://devcenter.heroku.com/articles/dyno-metadata).

**Pull request merged**